src/prometheus.pest

```diff
@@ -79,7 +79,7 @@ selector_range = @{ ws_opt ~ "[" ~ duration_value ~ duration_unit ~ "]" }
 metric_selector = { selector_metric ~ selector_label? }
 
 offset = ${ duration_value ~ duration_unit }
-offset_clause = _{ ws_req ~ ^"offset" ~ ws_req ~ offset }
+offset_clause = _{ ws_opt ~ ^"offset" ~ ws_req ~ offset }
```

Comment:
Unrelated to the `bool` modifier change, but I also noticed this is supported and Prometheus can parse it but was failing with the library.
```
promtool check rules /dev/stdin <<EOF                                                                                                                                                                   
groups:
- name: test
  rules:
  - alert: NoSpaceBeforeOffset
    expr: avg_over_time(prometheus_sd_discovered_targets[15m]offset 1h)
EOF
Checking /dev/stdin
  SUCCESS: 1 rules found
```